### PR TITLE
STM: Added cruising state and enforced kReady for all modules before acceleration

### PR DESCRIPTION
**Pull request by mifrandir**
This implements a good sketch of how we could implement a cruising state without affecting the current behaviour.

## Changes include:
- added Cruising state internally
- added `kCruising` to central data structure
- added maximum_velocity to navigation data
- refactoring in Navigation and Sensors subsystems

## TODO
- [x] all modules need to handle `kCruising`
  - [x] MTC
  - [x] NAV
  - [x] TLM
  - [x] SNS
- [x] batteries, telemetry and sensors need to set `kReady` while in `kCalibrating`
- [x] check with tech heads that this is the desired behaviour
- [x] add calibration stage for sensors (batteries in particular)
- [x] fix navigation 

## Caveats

While this PR aims to solve problems, it's not without its own shortcomings. In particular, we were able to adapt *most* parts of the code base to the new Cruising state. The problem that remains is that the fake sensors, in particular, the fake keyence are not able to handle this change. Unfortunately, fixing this is way beyond the scope of this already massive PR (and quite probably the current generation of the HYPED Software Team). Therefore, I have decided to change the maximum velocity in such a way that the simulation using the most recent fake sensor readings will never enter the Cruising state.

### CODE REVIEW

**Comment by mifrandir** [comment]
Here's the corresponding state diagram:

stm_diagram_v1 1 1

File: src/state_machine/state.hpp [hunk 1]
```diff
@@ -104,6 +104,7 @@ MAKE_STATE(Idle)            // State on startup
 MAKE_STATE(Calibrating)     // Calibrating starts after user input is given
 MAKE_STATE(Ready)           // After calibration has finished
 MAKE_STATE(Accelerating)    // First phase of the run
+MAKE_STATE(Cruising)        // Itermediate phase to not exceed maximum velocity
```
Comment (DonaldJennings) [inline src/state_machine/state.hpp]: Typo with intermediate

File: src/state_machine/transitions.hpp [hunk 2]
```diff
@@ -65,9 +65,10 @@ bool checkModulesInitialised(Logger &log, EmergencyBrakes embrakes_data, Navigat
                              Sensors sensors_data, Motors motors_data);
 
 /*
- * @brief    Returns true iff all of Embrakes, Navigation and Motors are ready.
+ * @brief    Returns true iff all modulesare ready.
```
Comment (DonaldJennings) [inline src/state_machine/transitions.hpp]: Typo with "modulesare" to "modules are". The meaning is clear from context but just a consistency thing.

**Review by DonaldJennings — CHANGES_REQUESTED** [review]
Very minor typos. Otherwise looks all good

**Review by Akshath29 — APPROVED** [review]

**Comment by mifrandir** [comment]
The only thing we're waiting for at the moment is sensors to add their calibration.

File: src/data/data.cpp [hunk 3]
```diff
@@ -29,9 +29,9 @@ const char* states[num_states] = {
   "Calibrating",
   "Ready",
   "Accelerating",
+  "Cruising",
   "NominalBraking",
   "EmergencyBraking",
-  "RunComplete",
   "FailureStopped",
   "Exiting",
```
Comment (robertasn) [inline src/data/data.cpp]: Do we still need this one here?

**Review by robertasn — CHANGES_REQUESTED** [review]

File: src/data/data.cpp (hunk 3 above)
Comment (mifrandir) [reply src/data/data.cpp]: No, good spot

**Review by robertasn — CHANGES_REQUESTED** [review]
Please resolve conflicts

**Review by robertasn — CHANGES_REQUESTED** [review]
image

**Comment by mifrandir** [comment]
> image

@MrTooOldDriver @Videosquared

**Comment by robertasn** [comment]
image
Does not exit NOMINAL_BRAKING state. In addition, CRUISING state is never entered.

**Review by Akshath29 — APPROVED** [review]

**Comment by robertasn** [comment]
A pull request has been merged into develop, which had unit tests for telemetry. Could you merge in develop and add new tests for the cruising part?

**Review by DonaldJennings — APPROVED** [review]
Looks good to me!

**Review by robertasn — APPROVED** [review]
LGTM

**Pull request merged**